projects/plugins/boost/app/assets/src/js/pages/critical-css-advanced/critical-css-advanced.tsx

```diff
@@ -0,0 +1,96 @@
+import { __, _n, sprintf } from '@wordpress/i18n';
+import {
+	replaceCssState,
+	updateProvider,
+} from '$features/critical-css/lib/stores/critical-css-state';
+import { groupErrorsByFrequency } from '$features/critical-css/lib/stores/critical-css-state-errors';
+import { type Provider } from '$features/critical-css/lib/stores/critical-css-state-types';
+import { navigate } from '$lib/utils/navigate';
+import { BackButton, CloseButton } from '$features/ui';
+import CriticalCssErrorDescription from '$features/critical-css/error-description/error-description';
+import InfoIcon from '$svg/info';
+import styles from './critical-css-advanced.module.scss';
+
+type PropTypes = {
+	issues: Provider[];
+};
+
+const AdvancedCriticalCss: React.FC< PropTypes > = ( { issues } ) => {
+	/**
+	 * Automatically navigate back to main Settings page if generator isn't done.
+	 */
+	if ( issues.length === 0 ) {
+		navigate( '/' );
+	}
+
+	const dismissedIssues = issues.filter( issue => issue.error_status === 'dismissed' );
+	const activeIssues = issues.filter( issue => issue.error_status !== 'dismissed' );
+
+	const heading =
+		activeIssues.length === 0
+			? __( 'Congratulations, you have dealt with all the recommendations.', 'jetpack-boost' )
+			: __(
+					'While Jetpack Boost has been able to automatically generate optimized CSS for most of your important files & sections, we have identified a few more that require your attention.',
+					'jetpack-boost'
+			  );
+
+	function showDismissedIssues() {
+		replaceCssState( {
+			providers: issues.map( issue => {
+				issue.error_status = 'active';
+				return issue;
+			} ),
+		} );
+	}
+
+	function dismissProvider( { key }: Provider ) {
+		updateProvider( key, { error_status: 'dismissed' } );
+	}
```

Comment:
wow, I'm kind of surprised this works 😅

Reply:
Now that you mention it - I feel like there's something off here. We are doing server requests after all.

I'm kind of fuzzy on how to use hooks in react, but I feel this is somewhat wrong.

The child component is basically updating issues directly, and then the parent component passes down the updated list and react re-renders the child component?

Maybe this should be react data store instead?